libs/qCC_io/LASFilter.cpp

```diff
@@ -32,1408 +32,1167 @@
+
+	//by default we read colors as triplets of 8 bits integers but we might dynamically change this
+	//if we encounter values using 16 bits (16 bits is the standard!)
+	unsigned char colorCompBitShift = 0;
+	bool forced8bitRgbMode = s_lasOpenDlg->forced8bitRgbMode();
+	ColorCompType rgb[3] = { 0, 0, 0 };
+
+	IdList extraFieldsToLoad;
+	StringList extraNamesToLoad;
+	for (unsigned i = 0; i < extraDimensionsIds.size(); ++i)
+	{
+		if (s_lasOpenDlg->doLoadEVLR(i)) {
+			extraFieldsToLoad.push_back(extraDimensionsIds[i]);
+			extraNamesToLoad.push_back(extraDimensions[i]);
 		}
+	}
 
-		//dialog to choose the fields to load
-		if (!s_lasOpenDlg)
+    bool tiling = s_lasOpenDlg->tileGroupBox->isChecked();
+   
+
+    QScopedPointer<ccProgressDialog> pDlg(0);
+    if (parameters.parentWidget)
+    {
+        pDlg.reset(new ccProgressDialog(true, parameters.parentWidget)); //cancel available
+        pDlg->setMethodTitle(QObject::tr("Open LAS file"));
+        pDlg->setInfo(QObject::tr("Points: %1").arg(nbOfPoints));
+        pDlg->start();
+    }
+
+	if (tiling)
+    {
+		Tiler tiler;
+		PointTable table;
+		PointViewSet pointViewSet;
+
+        // tiling (vertical) dimension
+        unsigned vertDim = 2;
+        switch (s_lasOpenDlg->tileDimComboBox->currentIndex())
+        {
+        case 0: //XY
+            vertDim = 2;
+            break;
+        case 1: //XZ
+            vertDim = 1;
+            break;
+        case 2: //YZ
+            vertDim = 0;
+            break;
+        default:
+            assert(false);
+            break;
+        }
+
+        auto w = static_cast<unsigned>(s_lasOpenDlg->wTileSpinBox->value());
+        auto h = static_cast<unsigned>(s_lasOpenDlg->hTileSpinBox->value());
+
+        QString outputBaseName = s_lasOpenDlg->outputPathLineEdit->text() + "/" + QFileInfo(filename).baseName();
+        if (!tiler.init(w, h, vertDim, outputBaseName, bbMin, bbMax, table))
+        {
+            return CC_FERR_NOT_ENOUGH_MEMORY;
+        }
+
+
+		PointViewSet viewSet;
+		PointViewPtr pointView;
+
+		auto prepareAndExecture = [&lasReader, &table]() -> PointViewSet {
+			lasReader.prepare(table);
+			lasReader.prepare(table);
+			return lasReader.execute(table);
+		};
+
+		QFuture<PointViewSet> reader = QtConcurrent::run(prepareAndExecture);
+		if (parameters.parentWidget)
 		{
-			s_lasOpenDlg = QSharedPointer<LASOpenDlg>(new LASOpenDlg());
+			pDlg.reset(new ccProgressDialog(false, parameters.parentWidget));
+			pDlg->setMethodTitle(QObject::tr("LAS file"));
+			pDlg->setInfo(QObject::tr("Please wait... reading in progress"));
+			pDlg->setRange(0, 0);
+			pDlg->setModal(true);
+			pDlg->start();
 		}
-		s_lasOpenDlg->setDimensions(dimensions);
-		s_lasOpenDlg->clearEVLRs();
-		s_lasOpenDlg->setInfos(filename, nbOfPoints, bbMin, bbMax);
-		if (extraDimension)
+		
+		while (!reader.isFinished())
 		{
-			assert(!evlrs.empty());
-			for (const EVLR& evlr : evlrs)
+#if defined(CC_WINDOWS)
```

Comment:
Is there a way around using Sleep/usleep? To me this is a red flag that something isn't implemented properly w.r.t. threads or concurrency.

Is there a way to pass in a callback to the reader to handle events processing and then simply use `reader.waitForFinished()` here?